### PR TITLE
Updates Closure Compiler & Library to latest versions.

### PR DESCRIPTION
**Pull request by gkdn**
This also updates protobuf and Closure template dependencies.

The patch removes //closure/library package since Closure Library recently
updated to include necessary BUILD targets under @com_google_javascript_closure_library//closure/goog.

For migration:
Most users can do replace @io_bazel_rules_closure//closure/library with
@com_google_javascript_closure_library//closure/goog.
Note that :all_js targets are not provided and can be replaced with specific
targets in the particular package.

Co-authored-by: pcj

### CODE REVIEW

**Comment by gkdn** [comment]
@jDramaix @pcj 
There are couple of changes in flight to Closure Library. After that all tests will start passing but we can start the review process.

File: closure/repositories.bzl [hunk 1]
```diff
@@ -744,23 +744,24 @@ def com_google_protobuf():
 def com_google_protobuf_js():
     http_archive(
         name = "com_google_protobuf_js",
-        strip_prefix = "protobuf-3.14.0/js",
-        sha256 = "d0f5f605d0d656007ce6c8b5a82df3037e1d8fe8b121ed42e536f569dec16113",
+        build_file = "@io_bazel_rules_closure//closure/protobuf:protobuf_js.BUILD",
+        sha256 = "9111bf0b542b631165fadbd80aa60e7fb25b25311c532139ed2089d76ddf6dd7",
+        strip_prefix = "protobuf-3.18.1/js",
```
Comment (jDramaix) [inline closure/repositories.bzl:749]: Last version is 3.19.1 see https://github.com/bazelbuild/rules_closure/pull/539/files

**Comment by pcj** [comment]
Awesome @gkdn, thanks for carrying this forward!  Very excited about this.

File: closure/repositories.bzl (hunk 1 above)
Comment (gkdn) [reply closure/repositories.bzl:749]: Looking at the PR, it seems like it will pull in new deps and that will require additional changes.

**Comment by gkdn** [comment]
@jDramaix PTAL

File: closure/testing/test/dom_test.js [hunk 2]
```diff
@@ -47,4 +47,4 @@ function testGetTextContent() {
 function testHtml() {
   assertHTMLEquals('<div id="hello">Hello World!</div>',
                    goog.dom.getRequiredElement('hello').outerHTML);
-}
+}
```
Comment (jDramaix) [inline closure/testing/test/dom_test.js:50]: missing new line

**Review by jDramaix — APPROVED** [review]

File: closure/repositories.bzl [hunk 3]
```diff
@@ -707,16 +705,11 @@ def com_google_javascript_closure_compiler():
...
-        name = "com_google_javascript_closure_library",
-        urls = [
-            "https://mirror.bazel.build/github.com/google/closure-library/archive/v20191027.tar.gz",
-            "https://github.com/google/closure-library/archive/v20191027.tar.gz",
-        ],
-        sha256 = "d21fdefb72d88a1aa629279c53a95fdd4d5632d0e5b13e6b591b177d25bde337",
-        strip_prefix = "closure-library-20191027",
-        build_file = str(Label("//closure/library:closure_library.BUILD")),
+       name = "com_google_javascript_closure_library",
+       sha256 = "",
```
Comment (pcj) [inline closure/repositories.bzl:710]: missing sha256

**Review by pcj — APPROVED** [review]

File: closure/repositories.bzl (hunk 3 above)
Comment (gkdn) [reply closure/repositories.bzl]: Good catch

File: closure/testing/test/dom_test.js (hunk 2 above)
Comment (gkdn) [reply closure/testing/test/dom_test.js]: Done.

**Pull request merged**